lib/lintPostcssResult.mjs

```diff
@@ -88,28 +78,37 @@ export default async function lintPostcssResult(stylelintOptions, postcssResult,
 
 		// Log the rule's severity in the PostCSS result
 		const defaultSeverity = config.defaultSeverity || DEFAULT_SEVERITY;
-		// disableFix in secondary option
-		const disableFix = (secondaryOptions && secondaryOptions.disableFix === true) || false;
 
 		postcssResult.stylelint.ruleSeverities[ruleName] =
 			(secondaryOptions && secondaryOptions.severity) || defaultSeverity;
 		postcssResult.stylelint.customMessages[ruleName] = secondaryOptions && secondaryOptions.message;
 		postcssResult.stylelint.customUrls[ruleName] = secondaryOptions && secondaryOptions.url;
 		postcssResult.stylelint.ruleMetadata[ruleName] = ruleFunction.meta || {};
 
+		const shouldWarn = ruleFunction.meta?.fixable && !stylelintOptions.quietDeprecationWarnings;
+		const disableFix = secondaryOptions?.disableFix === true;
+		const fix = !disableFix && config.fix && isFixCompatible(postcssResult, ruleName);
+		const context = {
+			...ctx,
+			// context.fix is unlikely to be removed in the foreseeable future
+			// due to the sheer number of rules in the wild that rely on it
+			get fix() {
+				if (shouldWarn) {
+					emitDeprecationWarning(
```

Comment:
@ybiquitous should this be called once per lint?
i.e. only for the first affected rule

The consequence will be that once the rule is fixed it might surface another warning for a subsequent rule.